lib/auth0/appEndpointHandler.js

```diff
@@ -6,11 +6,15 @@ module.exports = function (webtask, options, ctx, req, res, routingInfo) {
     function authenticate() {
         var token = options.getAccessToken(ctx, req);
         if (!token) {
+            var stateInfo = {
+                appPath: routingInfo.appPath,
+                query: routingInfo.query
+            };
             return options.loginError({
                 code: 401,
                 message: 'Unauthorized.',
                 error: 'Missing access token.',
-                redirect: routingInfo.baseUrl + '/login'
+                redirect: routingInfo.baseUrl + '/login?state=' + encodeURIComponent(JSON.stringify(stateInfo))
```

Comment:
A related IETF draft is described at [https://tools.ietf.org/html/draft-ietf-oauth-jwsreq-09](https://tools.ietf.org/html/draft-ietf-oauth-jwsreq-09). In this draft, all of the query parameters in an OAuth 2.0 authorization request can be encoded in a JWT, not just the `state` parameter.